sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala

```diff
@@ -44,18 +44,18 @@ class RegexParser(pattern: String) {
   private var pos = 0
 
   def parse(): RegexAST = {
-    val ast = parseInternal()
+    val ast = parseUntil(() => eof())
     if (!eof()) {
       throw new RegexUnsupportedException("failed to parse full regex")
     }
     ast
   }
 
-  private def parseInternal(): RegexAST = {
-    val term = parseTerm(() => peek().contains('|'))
+  private def parseUntil(until: () => Boolean): RegexAST = {
+    val term = parseTerm(() => until() || peek().contains('|'))
     if (!eof() && peek().contains('|')) {
```

Comment:
Nit: Not needed as part of this PR, but I think `peek()` will return `None` at EOF, so this can be simplified to:
```suggestion
    if (peek().contains('|')) {
```
Applies to a few other places where `eof` and `peek` are combined.